packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -15,7 +15,31 @@ const double _kCloseProgressThreshold = 0.5;
 const Color _kTransparent = const Color(0x00000000);
 const Color _kBarrierColor = Colors.black54;
 
+/// A material design bottom sheet.
+///
+/// There are two kinds of bottom sheets in material design:
+///
+///  * _Persistent_. A persistent bottom sheet shows information that
+///    supplements the primary content of the app. A persistent bottom sheet
+///    remains visible even when the user interacts with other parts of the app.
+///    Persistent bottom sheets can be created with the
+///    [Scaffold.showBottomSheet] function.
+///
+///  * _Modal_. A modal bottom sheet is an alternative to a menu or a dialog and
+///    prevents the user from interacting with the rest of the app. Modal bottom
+///    sheets can be created with the [showModalBottomSheet] function.
+///
+/// The [BottomSheet] widget itself is rarely used directly. Instead, prefer to
+/// create a persistent bottom sheet with [Scaffold.showBottomSheet] and a modal
+/// bottom sheet with [showModalBottomSheet].
+///
+/// See also:
+///
+///  * [Scaffold.showBottomSheet]
+///  * [showModalBottomSheet]
+///  * <https://www.google.com/design/spec/components/bottom-sheets.html>
 class BottomSheet extends StatefulWidget {
+  /// Creates a bottom sheet.
```

Comment:
too vague. add notes about some arguments of interest, or say who calls this normally